text/3723-static-function-argument-unpacking.md

```diff
@@ -0,0 +1,753 @@
+- Feature Name: `static_function_argument_unpacking`
+- Start Date: 2024-10-30
+- RFC PR: [rust-lang/rfcs#3723](https://github.com/rust-lang/rfcs/pull/3723)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC adds call-site unpacking of tuples, tuple structs, and fixed-size arrays, using `...expr` within the function call's parentheses as a shorthand for passing arguments. The full contents of these collections with known sizes are unpacked directly as the next arguments of a function call, desugaring into the corresponding element accesses during compilation.
+
+# Motivation
+[motivation]: #motivation
+
+Argument unpacking reduces the verbosity and increases the ergonomics of Rust, it:
+
+- Improves code writing ergonomics by removing the need for repetitive, unneeded intermediate steps.
+- Allows more concise code in terms of number of lines and, occasionally, line length.
+- Allows reducing the number of named local variables in scope.
+- Is intuitive for developers accustomed to argument unpacking from other programming languages.
+- Adds a missing piece to the family of certain kind of syntactic sugar already in Rust, with features such as *struct update syntax* and *destructuring assignment*.
```

Comment:
Struct update syntax and destructuring assignment have the same type on both sides. The sequence of parameters of a function (or its subsequence) is rarely identical to another type, except in the niche cases of pairs (e.g. a function that accepts two parameters of the same type and compares them) and well-defined vector types (e.g. `(x: f32, y: f32, z: f32)` or `(r: u8, g: u8, b: u8)`. In the former case the function should accept `[T; 2]` instead, and in the latter case the function should just define a type to avoid the boilerplate.